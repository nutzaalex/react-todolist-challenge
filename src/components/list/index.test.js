import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import List from './';

const todoList = [{id: 0, item: 'test1'}, {id: 1, item: 'test2'}, {id: 2, item: 'test3'}];

const spy = jest.fn();

describe('List', () => {
    it('matches snapshot', () => {
        const comp = renderer.create(<List todoList={todoList} deleteItem={jest.fn} updateItem={jest.fn}/>);

        expect(comp.toJSON()).toMatchSnapshot();
    });
    beforeEach(() => {
        window.prompt = spy;
    });
    it('calls deleteItem', () => {
        const deleteItemSpy = jest.fn();
        const comp = mount(<List todoList={todoList} deleteItem={deleteItemSpy} />);

        expect(deleteItemSpy).not.toHaveBeenCalled();
        comp.find('[type="delete"]').at(1).simulate('click');
        expect(deleteItemSpy).toHaveBeenCalledTimes(1);
        expect(deleteItemSpy).toHaveBeenCalledWith(todoList[1]);
    });
    it('opens a prompt to edit the todo', () => {
        const updateItemSpy = jest.fn();
        const comp = mount(<List todoList={todoList} updateItem={updateItemSpy} />);

        expect(updateItemSpy).not.toHaveBeenCalled();
        comp.find('[type="edit"]').at(1).simulate('click');
        expect(spy).toHaveBeenCalledTimes(1);
    });
});