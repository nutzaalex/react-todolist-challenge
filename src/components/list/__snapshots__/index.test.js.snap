// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List matches snapshot 1`] = `
<table>
  <thead>
    <tr>
      <th
        width="66%"
      >
        Item
      </th>
      <th>
        Actions
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        test1
      </td>
      <td>
        <button
          className="button edit"
          onClick={[Function]}
        >
          Edit
        </button>
        <button
          className="button delete"
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
    </tr>
    <tr>
      <td>
        test2
      </td>
      <td>
        <button
          className="button edit"
          onClick={[Function]}
        >
          Edit
        </button>
        <button
          className="button delete"
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
    </tr>
    <tr>
      <td>
        test3
      </td>
      <td>
        <button
          className="button edit"
          onClick={[Function]}
        >
          Edit
        </button>
        <button
          className="button delete"
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
